### PR TITLE
Make Averager serializable and constructible from options. 

### PR DESCRIPTION
**Pull request by osheamonn**
## Proposed changes

Adds functionality to the averager that I found I needed for the control systems. 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ControlSystem/Averager.cpp
Comment (knelli2) [inline src/ControlSystem/Averager.cpp]: Should you also check `raw_qs_`? I don't think there can be a case where all the timescales, weights, averaged values, etc. are the same and the `raw_qs_` are different, but I'm not 100% sure.

File: tests/Unit/ControlSystem/Test_Averager.cpp
Comment (knelli2) [inline tests/Unit/ControlSystem/Test_Averager.cpp]: Included `TestHelpers` twice.

File: tests/Unit/ControlSystem/Test_Averager.cpp
Comment (knelli2) [inline tests/Unit/ControlSystem/Test_Averager.cpp]: You should also check that two averagers are (not)equal after an update. Here you're just checking that the default values of most members are the same.

Maybe also add a few `CHECK_FALSE`s as well.

File: src/ControlSystem/Averager.cpp
Comment (knelli2) [inline src/ControlSystem/Averager.cpp]: Doxygen doesn't work in cpp files.

File: src/ControlSystem/Averager.hpp
Comment (knelli2) [inline src/ControlSystem/Averager.hpp]: Clang-tidy complaining about duplicates. Maybe because they are public?

**Review by knelli2 — CHANGES_REQUESTED** [review]

File: src/ControlSystem/Averager.hpp
Comment (wthrowe) [inline src/ControlSystem/Averager.hpp:13]: Alphabetize.

File: src/ControlSystem/Averager.hpp
Comment (wthrowe) [reply src/ControlSystem/Averager.hpp]: It's a bug.  Just NOLINT them.

File: src/ControlSystem/Averager.hpp
Comment (wthrowe) [inline src/ControlSystem/Averager.hpp]: This one doesn't need to be a friend.

File: src/ControlSystem/Averager.hpp
Comment (wthrowe) [inline src/ControlSystem/Averager.hpp]: Should probably put something here.

File: tests/Unit/ControlSystem/Test_Averager.cpp
Comment (wthrowe) [inline tests/Unit/ControlSystem/Test_Averager.cpp]: Change the test cases in this file into functions and call them all from one test case.  We realized at some point that each test case has a significant overhead on some clusters.

File: src/ControlSystem/Averager.hpp
Comment (wthrowe) [inline src/ControlSystem/Averager.hpp:11]: Merge with above block of includes.

File: src/ControlSystem/Averager.cpp
Comment (wthrowe) [inline src/ControlSystem/Averager.cpp:224]: This isn't new, but do you know why this is instantiated only for 2?

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Review by wthrowe — DISMISSED** [review]
I'm happy for a squash, but wait for @knelli2's approval.

**Review by knelli2 — DISMISSED** [review]
LGTM :)

**Review by knelli2 — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**
